File: androidApp/src/main/kotlin/br/com/qualorock/androidApp/ui/components/ConsentCapture.kt
package br.com.qualorock.androidApp.ui.components

import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Checkbox
import androidx.compose.material3.CheckboxDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import design.NightlifeGvTokens

const val ConsentCheckboxTestTag = "consentCheckbox"

/**
 * A5 — required, non-pre-checked consent acceptance (AUTH-02/AUTH-03), shared shape reused
 * across signup screens. `accepted` is hoisted — the caller owns whether it's persisted/reset.
 */
@Composable
fun ConsentCapture(accepted: Boolean, onAcceptedChange: (Boolean) -> Unit, modifier: Modifier = Modifier) {
    Row(
        verticalAlignment = Alignment.CenterVertically,
        modifier = modifier.padding(vertical = NightlifeGvTokens.Space2Dp.dp),
    ) {
        Checkbox(
            checked = accepted,
            onCheckedChange = onAcceptedChange,
            colors = CheckboxDefaults.colors(checkedColor = Color(NightlifeGvTokens.AccentPink)),
            modifier = Modifier.testTag(ConsentCheckboxTestTag),
        )
        Text(
            text = "Li e aceito os Termos de Uso e a Política de Privacidade.",
            color = Color(NightlifeGvTokens.ColorTextSecondary),
            fontSize = NightlifeGvTokens.TextBody.SizeSp.sp,
        )
    }
}

File: iosApp/iosApp/UI/Components/ConsentCapture.swift
import SwiftUI
import shared

/// I5 — required, non-pre-checked consent acceptance (AUTH-02/AUTH-03), shared shape reused
/// across signup screens. `accepted` is hoisted — the caller owns whether it's persisted/reset.
struct ConsentCapture: View {
    @Binding var accepted: Bool

    var body: some View {
        HStack(alignment: .center, spacing: QorSpace.space2) {
            Toggle(isOn: $accepted) { EmptyView() }
                .labelsHidden()
                .toggleStyle(ConsentCheckboxToggleStyle())
                .accessibilityIdentifier("consent_checkbox")

            Text(String(localized: "consent_terms_text"))
                .font(.system(size: CGFloat(QualORockThemeTokens.TextBody.shared.SizeSp)))
                .foregroundStyle(QorColor.textSecondary)
        }
        .padding(.vertical, QorSpace.space2)
    }
}

/// A checkbox (square) rather than the default switch, matching Android's `Checkbox`.
private struct ConsentCheckboxToggleStyle: ToggleStyle {
    func makeBody(configuration: Configuration) -> some View {
        Button {
            configuration.isOn.toggle()
        } label: {
            Image(systemName: configuration.isOn ? "checkmark.square.fill" : "square")
                .foregroundStyle(configuration.isOn ? QorColor.accentPink : QorColor.textSecondary)
        }
        .buttonStyle(.plain)
    }
}
